core/src/Cuda/Kokkos_Cuda_Parallel_MDRange.hpp

```diff
@@ -95,11 +95,32 @@ class ParallelFor<FunctorType, Kokkos::MDRangePolicy<Traits...>, Kokkos::Cuda> {
 
   inline void execute() const {
     if (m_rp.m_num_tiles == 0) return;
-    const auto maxblocks = m_rp.space().cuda_device_prop().maxGridSize;
+    const auto maxblocks  = m_rp.space().cuda_device_prop().maxGridSize;
+    const auto maxthreads = m_rp.space().cuda_device_prop().maxThreadsDim;
+    [[maybe_unused]] const auto maxThreadsPerBlock =
+        m_rp.space().cuda_device_prop().maxThreadsPerBlock;
+    // make sure the Z dimension (it is less than x,y limits) isn't exceeded
+    const auto validateZ = [&](const int input) {
+      return (input > maxthreads[2] ? maxthreads[2] : input);
+    };
+    // make sure the block dimensions don't exceed the max number of threads
+    // allowed
+    const auto check_block_sizes = [&]([[maybe_unused]] const dim3& block) {
```

Comment:
I believe dim3 is a vector of uint

Reply:
I wonder if that implicitly converts int[3] to unsigned integers...

Reply:
I could cast it at the comparison. According to https://docs.nvidia.com/cuda/cuda-runtime-api/structcudaDeviceProp.html#structcudaDeviceProp_192d195493a9d36b2d827aaf3ffd89f1e, maxThreadsDim is type `int[3]`.